applications/GeoMechanicsApplication/custom_processes/find_neighbour_elements_of_conditions_process.cpp

```diff
@@ -8,406 +8,193 @@
+    default:
+        return false;
     }
-
-    return FacesMap.end();
-
-    KRATOS_CATCH("")
 }
 
-hashmap::iterator FindNeighbourElementsOfConditionsProcess::FindFaceReorderingHexahedra3D8(DenseVector<int> FaceIds,
-                                                                                           hashmap& FacesMap) const
+bool FindNeighbourElementsOfConditionsProcess::AreLinearRotatedEquivalents(std::vector<std::size_t> First,
+                                                                           const std::vector<std::size_t>& rSecond)
 {
-    KRATOS_TRY
-
-    hashmap::iterator itFace = FacesMap.find(FaceIds);
-    if (itFace != FacesMap.end()) return itFace;
-
-    if (FaceIds.size() == 4) {
-        // first try
-        std::swap(FaceIds[0], FaceIds[1]);
-        std::swap(FaceIds[1], FaceIds[2]);
-        std::swap(FaceIds[2], FaceIds[3]);
-
-        itFace = FacesMap.find(FaceIds);
-        if (itFace != FacesMap.end()) return itFace;
-
-        // Second try
-        std::swap(FaceIds[0], FaceIds[1]);
-        std::swap(FaceIds[1], FaceIds[2]);
-        std::swap(FaceIds[2], FaceIds[3]);
-
-        itFace = FacesMap.find(FaceIds);
-        if (itFace != FacesMap.end()) return itFace;
-
-        // Third try
-        std::swap(FaceIds[0], FaceIds[1]);
-        std::swap(FaceIds[1], FaceIds[2]);
-        std::swap(FaceIds[2], FaceIds[3]);
-
-        itFace = FacesMap.find(FaceIds);
-        if (itFace != FacesMap.end()) return itFace;
-    }
-
-    return FacesMap.end();
-
-    KRATOS_CATCH("")
+    const auto amount_of_needed_rotations = std::ranges::find(First, rSecond[0]) - First.begin();
+    std::rotate(First.begin(), First.begin() + amount_of_needed_rotations, First.end());
+    return First == rSecond;
 }
 
-hashmap::iterator FindNeighbourElementsOfConditionsProcess::FindFaceReorderingHexahedra3D20(DenseVector<int> FaceIds,
-                                                                                            hashmap& FacesMap) const
+bool FindNeighbourElementsOfConditionsProcess::AreQuadraticRotatedEquivalents(std::vector<std::size_t> First,
+                                                                              const std::vector<std::size_t>& rSecond)
 {
-    KRATOS_TRY
-
-    hashmap::iterator itFace = FacesMap.find(FaceIds);
-    if (itFace != FacesMap.end()) return itFace;
-
-    if (FaceIds.size() == 8) {
-        // first try
-        std::swap(FaceIds[0], FaceIds[1]);
-        std::swap(FaceIds[1], FaceIds[2]);
-        std::swap(FaceIds[2], FaceIds[3]);
-        std::swap(FaceIds[4], FaceIds[5]);
-        std::swap(FaceIds[5], FaceIds[6]);
-        std::swap(FaceIds[6], FaceIds[7]);
+    const auto amount_of_needed_rotations = std::ranges::find(First, rSecond[0]) - First.begin();
 
-        itFace = FacesMap.find(FaceIds);
-        if (itFace != FacesMap.end()) return itFace;
+    std::rotate(First.begin(), First.begin() + amount_of_needed_rotations, First.begin() + First.size() / 2);
```

Comment:
To make reading a little easier, we may want to introduce an iterator that points to the first mid-side node:
```suggestion
    auto first_mid_side_node_id = First.begin() + First.size() / 2;
    std::rotate(First.begin(), First.begin() + amount_of_needed_rotations, first_mid_side_node_id);
```
This variable can then also be reused by the next `std::rotate` invocation.

Reply:
Done